### PR TITLE
Add libgsf

### PR DESCRIPTION
**Pull request by sebastian-luna-valero**
Checklist

- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vend other packages
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/libgsf/build.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -euo pipefail
+
+# fix perl shebang
+sed -i.bak '1 s|^.*$|#!/usr/bin/env perl|g' $SRC_DIR/tests/t*.pl
+
+./configure --prefix=${PREFIX}
+make
```
Comment (marcelotrevisani) [inline recipes/libgsf/build.sh]: You might compile it in parallel
```suggestion
make -j${CPU_COUNT}
```

File: recipes/libgsf/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    - perl
+    - perl-xml-parser
+    - pkg-config
+    - {{ compiler('c') }}
+  host:
+    - glib
+    - libxml2
+    - zlib
+    - bzip2
+    - pcre 8.42*
```
Comment (marcelotrevisani) [inline recipes/libgsf/meta.yaml]: Should it be just ``8.42`` or maybe greater than?
```suggestion
    - pcre 8.42
```

Comment (sebastian-luna-valero) [reply recipes/libgsf/meta.yaml]: All other pcre versions available gave me problems with local circleci testing.

File: recipes/libgsf/build.sh (hunk 1 above)
Comment (sebastian-luna-valero) [reply recipes/libgsf/build.sh]: Not the part that is most time consuming, but will add it. Thanks!

File: recipes/libgsf/meta.yaml (hunk 2 above)
Comment (marcelotrevisani) [reply recipes/libgsf/meta.yaml]: I see, could you please pin it then? Just remove the ``*`` and it should be fine
:)

Comment (marcelotrevisani) [reply recipes/libgsf/meta.yaml]: ah, you already did it!
:)

**Comment by marcelotrevisani** [comment]
After that CI passes I think that is good to go! :tada:

**Pull request merged**